src/main/kotlin/com/terraformation/backend/accelerator/ParticipantProjectSpeciesService.kt

```diff
@@ -109,12 +131,74 @@ class ParticipantProjectSpeciesService(
     }
   }
 
+  /**
+   * When a species deliverable status is updated to "approved", we save a snapshot of the species
+   * list and reference it in the deliverable submission
+   */
+  @EventListener
+  fun on(event: DeliverableStatusUpdatedEvent) {
+    if (event.newStatus !== SubmissionStatus.Approved) {
+      return
+    }
+
+    val deliverable = deliverablesDao.fetchOneById(event.deliverableId)
+    if (deliverable == null || deliverable.deliverableTypeId != DeliverableType.Species) {
+      return
+    }
+
+    val speciesData =
+        participantProjectSpeciesStore.fetchSpeciesForParticipantProject(event.projectId)
+    val rows: List<List<Pair<String, String>>> =
+        speciesData.map {
+          listOf(
+              "Project ID" to it.project.id.toString(),
+              "Species ID" to it.species.id.toString(),
+              "Status" to it.participantProjectSpecies.submissionStatus.jsonValue,
+              "Rationale" to (it.participantProjectSpecies.rationale ?: ""),
```

Comment:
I believe CSVWriter will render nulls as zero-length strings, so maybe the `?: ""` isn't needed.